app/components/Settings/cron-picker/cron-helpers.js

```diff
@@ -9,3 +9,39 @@ export function mapHourToCronFormat(hour, amPm) {
 
     return hour;
 }
+
+export function getInitialCronMode(query) {
+  const { cronInterval, refreshInterval } = query;
+
+  if (cronInterval) {
+    if (cronInterval === '* * * * *') {
+      return 'MINUTE';
+    } else if (cronInterval === '*/5 * * * *') {
+      return 'FREQUENTLY';
+    } else if (cronInterval.match(/\S+? \* \* \* \*/)) {
+      return 'HOURLY';
+    } else if (cronInterval.match(/\S+? \S+? \* \* \*/)) {
+      return 'DAILY';
+    } else if (cronInterval.match(/\S+? \S+? \* \* \S+?/)) {
+      return 'WEEKLY';
+    } else if (cronInterval.match(/\S+? \S+? \S+? \* \*/)) {
+      return 'MONTHLY';
+    }
+  } else if (refreshInterval) {
+    if (refreshInterval <= 60) {
+      return 'MINUTE';
+    } else if (refreshInterval <= 5 * 60) {
+      return 'FREQUENTLY';
+    } else if (refreshInterval <= 60 * 60) {
+      return 'HOURLY';
+    } else if (refreshInterval <= 24 * 60 * 60) {
+      return 'DAILY';
+    } else if (refreshInterval <= 7 * 24 * 60 * 60) {
+      return 'WEEKLY';
+    } else if (refreshInterval <= 30 * 24 * 60 * 60) {
```

Comment:
The maximum refresh interval is 2147483647 (see [MDN}(https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout#Maximum_delay_value))